src/frontend/src/components/public/StatusChangeDropDown/populateModal.tsx

```diff
@@ -21,31 +24,61 @@ const PopulateModal = (
   pia: IPiaForm,
   nextStatus: string,
   changeStatusFn: ImodalCB,
+  from: string, // Added event variable whether this is from dropdown or submit behavior
+  time?: string, // Added optional time parameter passed for the auto-save failed modal
+  conflictUser?: string, // Added optional conflict user for the conflict modal
 ) => {
   const role = getUserRole();
   let useDefault = true;
   const currentStatus = pia.status || 'Completed';
+
+  // Handle autoSaveFailed modal with time
+  if (nextStatus === '_autoSaveFailed' && time) {
+    const autoSaveFailedModalWithTime = {
+      ...autoSaveFailedModal,
+      description: autoSaveFailedModal.description.replace('${time}', time),
+    };
+    changeStatusFn(autoSaveFailedModalWithTime, nextStatus, from);
+    return;
+  }
+
+  // Handle conflict modal with conflictUser
+  if (nextStatus === '_conflict' && conflictUser) {
+    const conflictModalWithUser = {
+      ...conflictModal, // Define a conflict modal
+      title: conflictModal.title.replace('${user}', conflictUser),
+      description: conflictModal.description.replace('${user}', conflictUser),
+    };
+    changeStatusFn(conflictModalWithUser, nextStatus, from);
+    return;
+  }
+
   if (
-    'changeStatus' in Object(statusList(pia)[currentStatus].Privileges)[role]
+    'changeStatus' in
+    Object(statusList(pia, from)[currentStatus].Privileges)[role]
   ) {
     if (
-      Object(statusList(pia)[currentStatus].Privileges)[role].changeStatus
+      Object(statusList(pia, from)[currentStatus].Privileges)[role].changeStatus
         .length !== 0
     ) {
-      Object(statusList(pia)[currentStatus].Privileges)[
+      Object(statusList(pia, from)[currentStatus].Privileges)[
         role
       ].changeStatus.forEach((status: ChangeStatus) => {
         if (status.status === nextStatus) {
           if (status.modal) {
```

Comment:
We should use guard statements and remove these heavily nested if statements.

Reply:
yes, I have removed such nested if statements and made changes.